### ISSUE TITLE
Can no longer connect to password-protected channels

### ISSUE THREAD

**Issue by mekolat** [reactiflux/discord-irc#332]
Prior to updating to `2.5.0` (was on `2.4.2`) my setup worked just fine, however now discord-irc refuses to connect to password-protected channels and I haven't changed anything in the settings.
The affected channel is `#hchat` on Rizon, I haven't tried other password-protected channels. The regular channels work just fine.

I get a `Illegal channel name` error message when debug is enabled:
```yaml
ERROR: Unhandled message: { prefix: 'magnet.rizon.net',
  server: 'magnet.rizon.net',
  command: '479',
  rawCommand: '479',
  commandType: 'normal',
  args:
   [ 'HercDiscord',
     '#hchat',
     '<password removed>',
     'Illegal channel name' ] }
```
<br>
<details><summary>full config file (click here)</summary>

```json
[
	{
		"nickname": "HercDiscord",
		"server": "irc.rizon.net",
		"discordToken": "<removed token>",
		"channelMapping": {
			"306632465294819331": "#Hercules",
			"306633733472256011": "##Hercules-script",
			"306634170858471425": "##Hercules-dev",
			"306634060438962207": "#Hercules-Testers",
			"306633825750876162": "#hchat <removed password>"
		},
		"ircOptions": {
			"userName": "HercDiscord",
			"realName": "Hercules Discord bridge",
			"password": "<removed password>",
			"port": 6697,
			"secure": true,
			"autoRejoin": true,
			"autoRenick": true
		},
		"format": {
			"commandPrelude": "Command sent by {$nickname}",
			"ircText": "<{$displayUsername}> {$text}",
			"urlAttachment": "<{$displayUsername}> {$attachmentURL}",
			"discord": "**<{$author}>** {$withMentions}"
		},
		"ignoreUsers": {
			"irc": [
				"Hercules"
			],
			"discord": []
		},
		"ircNickColor": false,
		"ircStatusNotices": false
	}
]
```

</details>

**Comment by mekolat** [comment]
Also:
```
(node:1434) ExperimentalWarning: The http2 module is an experimental API.
events.js:182
      throw er; // Unhandled 'error' event
      ^

Error: write EIO
    at _errnoException (util.js:1024:11)
    at WriteWrap.afterWrite [as oncomplete] (net.js:870:14)
```

**Comment by Throne3d** [comment]
Oops, sorry, I think this was broken upstream – I've started a fix [here](https://github.com/Throne3d/node-irc/pull/46) but still need to do a couple things, then I'll release a patch version that we should be able to get through here.

**Comment by Throne3d** [comment]
If you're able to try out the version of the code in PR #333, could you let me know if that solves the problem? You'll need to re-run `npm install` to make sure you have the latest version of the upstream dependency which caused this problem (`irc-upd`).

**Comment by mekolat** [comment]
works perfect, thanks

**Comment by Throne3d** [comment]
The pull request has now been merged, so I'll close this issue. Hopefully everything's fixed!

**Comment by LeonellS** [comment]
Will this fix be released as a patch on npm? `irc-upd` is still `0.8.0` there.
